src/renderer/components/Visualiser/store/actions.js

```diff
@@ -223,42 +223,53 @@ export default {
       const queryPattern = node.queryPattern;
       const queryPatternVariales = Array.from(new Set(queryPattern.match(/\$[^\s|)|;|,]*/g))).map(x => x.substring(1));
 
-      const explanationAnswers = (await node.explanation()).getAnswers();
-
-      const explanationPromises = [];
-      const explanationResult = [];
-
-      explanationAnswers.forEach((answer) => {
-        const answerVariabes = Array.from(answer.map().keys());
-
-        const isJointExplanation = answerVariabes.every(variable => queryPatternVariales.includes(variable));
-        if (answer.hasExplanation() && isJointExplanation) {
-          explanationPromises.push(answer.explanation());
-        } else if (!isJointExplanation) {
-          explanationResult.push(answer);
-        }
-      });
-
-      (await Promise.all(explanationPromises)).map(expl => expl.getAnswers()).reduce(collect, []).forEach((expl) => {
-        explanationResult.push(expl);
-      });
+      const explanation = await node.explanation();
 
-      if (explanationResult.length > 0) {
-        const data = await CDB.buildInstances(explanationResult);
-        const rpData = await CDB.buildRPInstances(explanationResult, data, false, graknTx);
-        data.nodes.push(...rpData.nodes);
-        data.edges.push(...rpData.edges);
-
-        state.visFacade.addToCanvas(data);
-        commit('updateCanvasData');
-        const nodesWithAttributes = await computeAttributes(data.nodes, graknTx);
+      const when = await (await explanation.getRule()).getWhen();
+      const isRelUnassigned = !when.match(/(\$[^\s]*|;|{)(\s*?\(.*?\))/)[1].includes('$');
```

Comment:
lol nice - can you write a unit test for this?

Reply:
thanks for asking for the test. turned out the implementation wasn't correct.

resolved in https://github.com/graknlabs/workbase/pull/298/commits/abf1d365c6d5789cb1b2440dda60a706361ebd6e

Reply:
excellente